tests/unit/test_middleware_options.py

```diff
@@ -1,10 +1,30 @@
 from unittest import TestCase
 
 from msgraphcore.middleware.options.auth_middleware_options import AuthMiddlewareOptions
+from msgraphcore.middleware.options.retry_middleware_options import RetryMiddlewareOptions
 
+retry_settings = {
+    'retry_total': 5,
+    'retry_backoff_factor': 0.8,
+    'retry_backoff_max': 100,
+    'retry_time_limit': 100,
+    'retry_on_status_codes': [502, 503],
+}
 
-class TestMiddlewareOptions(TestCase):
+
+class TestAuthMiddlewareOptions(TestCase):
     def test_multiple_scopes(self):
         graph_scopes = 'https://graph.microsoft.com/v1.0?scopes=mail.read%20user.read%20'
         auth_options = AuthMiddlewareOptions(graph_scopes)
         self.assertEqual(auth_options.scopes, graph_scopes)
+
+
+def test_retry_middleware_options():
```

Comment:
It's more effective to use integration tests to test middleware options and assert that they are changing the behaviour of the middleware.